sklearn/linear_model/stochastic_gradient.py

```diff
@@ -166,6 +167,19 @@ def _validate_params(self, set_max_iter=True, for_partial_fit=False):
                 # Before 0.19, default was n_iter=5
             max_iter = 5
         else:
+            if self.tol is None:
+                # max_iter was set, but tol wasn't. The docs / warning do not
+                # specify this case. In 0.20 tol would stay being None which
+                # is equivalent to -inf, but it will be changed to 1e-3 in
+                # 0.21. We warn users that the behaviour (and potentially
+                # their results) will change.
+                warnings.warn(
+                    "max_iter and tol parameters have been added in %s in "
+                    "0.19. If max_iter is set but tol is left unset, the "
+                    "default value for tol in 0.19 and 0.20 will be None "
+                    "but will change in 0.21 to 1e-3. Specify tol to "
+                    "silence this warning." % type(self).__name__,
+                    ChangedBehaviorWarning)
```

Comment:
I think this is technically meant to be a FutureWarning

Reply:
Right,ow it should be a FutureWarning since we didn't change yet.